### ISSUE TITLE
Press "F" twice to toggle "target follow" for the editor camera

### ISSUE THREAD

**Issue by lentsius-bark** [godotengine/godot-proposals#3164]
### Describe the project you are working on

An exploration themed 3D game set in a wholesome post apocalyptic world. The player controls a ship and fishes garbage out the ocean to help nature
There are paths that have cable-cars on path follows with custom animation for sway.

### Describe the problem or limitation you are having in your project

The issue is present whenever there is an object which requires
I want to follow the object as it is animated along the path, which can be done by holding F, but it becomes unintuitive when I want to press other keyboard shortcuts.
By having the option of camera following the object, I can easily tweak its animated properties to best fit the given situation and momentary physical implications that need to translate into an animation.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

If we could press "F" twice which would automatically make the editor camera follow target, we would help users with situations when an animated or simulated object leaves the camera view, giving constant visual feedback on what's going on.

At the moment the user needs to pause the animation or simulation (tool scripts) and refocus, or zoom out. 
Zooming out although a solution, may induce either loss of perceived visual details necessary in development for the given object or the distance may become so great that it no longer allows the user to properly assess their design.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

The users presses "F" twice instead of once, the camera zooms in on the target, selected node and follows it if its position changes.
- Rotating the view in editor doesn't cancel the follow
- moving (panning) the view does cancel the camera follow

Bonus: If this could be somehow linked with the feature which allows the editor camera to override the running game's one, I believe this could easily become a vital debug tool, where in-game users could tweak things that are in motion in editor, focus in on them and switch to the game instantly seeing the results, thus greatly enhancing the features usefulness.
i.e. 
- enemy AI, looking for the player while they're hidden. Tweak the AI logic, play with its parameters.
- a physical object dropping through an obstacle course
- objects that are following paths and are being tweaked

https://user-images.githubusercontent.com/38346332/130144331-a4c2c03c-42fa-4c7f-aee5-3d48e3fa100f.mp4

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I believe this could be an addon indeed.

### Is there a reason why this should be core and not an add-on in the asset library?

Moving the camera, focusing on objects is a core part of development that takes place in the editor at all times and becomes even more pronounced with 3D. This feature simply enhances the already existing functionality and promotes an easier way of working with objects in motion.

**Comment by Calinou** [comment]
That's a good idea. It's likely worth implementing in the 2D editor as well.

**Comment by lentsius-bark** [comment]
Yeah! Alt + F sounds good. I've used software that had either, Unity has a double F for instance I believe.

**Comment by GuyUnger** [comment]
couldn't it just work by pressing F once, and the moment you pan it's released. there wouldn't be any difference in behavior when focusing on an object that doesn't move, and if you focus on a moving object you probably want it to keep the focus by default, and if you don't it's easy to get out of it by padding a little bit

**Comment by lentsius-bark** [comment]
> couldn't it just work by pressing F once, and the moment you pan it's released. there wouldn't be any difference in behavior when focusing on an object that doesn't move, and if you focus on a moving object you probably want it to keep the focus by default, and if you don't it's easy to get out of it by padding a little bit

After a bit of thought, I don't think this would work well IMO, as a common workflow if to focus on an object an then move it right away. If the camera follow was to be automatic it would create a confusing situation where the user tries moving an object but it seems to be still or rather then entire world is moving!! 😄 

Yeah a shortcut should be just fine.

**Comment by GuyUnger** [comment]
what if moving the object also cancels the focus? maybe that sounds like patching up the problem but i think it could actually be intuitive

**Comment by GuyUnger** [comment]
If there's still a problem with that a shortcut might be necessary, but if the shortcut can be avoided it's less clutter in Godot and more people will benefit from this feature (since almost nobody ever finds these special shortcuts)

**Comment by me2beats** [comment]
This would be useful.
I would like to implement this as a GDScript plugin but I can't because there is no a way to set  the camera translation
https://github.com/godotengine/godot-proposals/issues/3287
